olive/cli/auto_opt.py

```diff
@@ -267,19 +304,49 @@ def _get_passes_config(self, config: Dict[str, Any], olive_config: OlivePackageC
             if value is not None:
                 set_nested_dict_value(passes_config, keys, value)
 
-        del passes_config["conversion" if self.args.use_model_builder else "model_builder"]
-        if (self.args.provider != "QNNExecutionProvider") or self.args.use_model_builder:
+        passes_to_remove = set()
+        if config["input_model"]["type"].lower() == "onnxmodel":
+            # remove passes that only operate on PyTorch/Hf models
+            passes_to_remove.update(["capture_split_info", "conversion", "model_builder", "split_model"])
+        else:
+            # only one graph capture pass is used
+            passes_to_remove.add("conversion" if self.args.use_model_builder else "model_builder")
+
+        # optional split passes
+        if self.args.num_splits is None and self.args.cost_model is None:
+            passes_to_remove.update(["capture_split_info", "split_model"])
+        if self.args.cost_model is not None and self.args.memory is None:
+            raise ValueError("memory is required if cost_model is provided.")
+
+        if self.args.provider != "QNNExecutionProvider":
             # Use the DynamicToFixedShape pass only for QNNExecutionProvider
             # becase QNN doesn't support dynamic shaped inputs
-            del passes_config["to_fixed_shape"]
-        elif self.args.provider != "JsExecutionProvider":
-            # JS EP doesn't support fp16
-            del passes_config["fp16_to_fp32"]
+            passes_to_remove.add("to_fixed_shape")
+        else:
+            # qnn ep might not supported optimized model
+            # will re-enable it if needed in the future
+            passes_to_remove.update(["transformer_optimizer", "optimizer"])
+
+        if self.args.provider != "JsExecutionProvider":
```

Comment:
There is also WebGPU EP which has the same limitations while working in the web browser env.

Reply:
Added webgpu to the condition too